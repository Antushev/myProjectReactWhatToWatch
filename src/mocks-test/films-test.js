import {generateFilm, generateFilms} from '../mocks/films.js';

const FILMS_NUMBER = 10;

const film = generateFilm(1);
const films = generateFilms(FILMS_NUMBER);

export {film, films};
