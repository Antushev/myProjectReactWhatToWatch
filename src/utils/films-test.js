const films = [
  {
    id: 1,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2001
  },
  {
    id: 2,
    name: `Интерстеллар`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2000
  },
  {
    id: 3,
    name: `Спасти рядоого Райана`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2004
  },
  {
    id: 4,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2014
  },
  {
    id: 5,
    name: `Власталин колец: Возвращение короля`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2014
  },
  {
    id: 6,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2019
  },
  {
    id: 7,
    name: `Хоббит: Нежданное путешествие`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2020
  },
  {
    id: 8,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2018
  }
];

const film = {
  id: 1,
  name: `Интерстеллар`,
  picture: `img/pictrue.jpg`,
  genre: `Adventure`,
  date: 2014
};

export {film, films};
