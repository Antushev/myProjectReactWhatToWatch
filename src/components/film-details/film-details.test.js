import React from 'react';
import renderer from 'react-test-renderer';

import {film, films} from '../../mocks-test/films-test.js';

import FilmDetails from './film-details.jsx';

const activeTab = `overview`;

describe(`FilmDetailsComponent`, () => {
  it(`FilmDetailsComponentSnapshot`, () => {
    const tree = renderer
      .create(
          <FilmDetails
            film={film}
            films={films}
            activeTab={activeTab}
            renderTabs={() => {}}
            handleFilmClick={() => {}}
          />, {
            createNodeMock: () => {
              return {};
            }
          }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
