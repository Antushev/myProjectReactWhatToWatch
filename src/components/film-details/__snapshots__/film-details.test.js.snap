// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetailsComponent FilmDetailsComponentSnapshot 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog"
  >
    <div
      className="catalog__movies-list"
    >
      <h1>
        Идёт загрузка
      </h1>
    </div>
  </section>
</div>
`;
